### ISSUE TITLE
OpenApi 3.0 query params format

### ISSUE THREAD

**Issue by redvex** [rswag/rswag#317]
Hello everyone, I'm trying to document my project with OpenAPI 🙂

This params in my spec allow me to pass the test:
```ruby
parameter name: :supply_centers_ids, in: :query, type: :array, collectionFormat: :csv, items: { type: :integer }
```

but it generates:
```yaml
parameters:
  - name: supply_centers_ids
    in: query
      collectionFormat: csv
      items:
        type: integer
      schema:
        type: array
```

that throw this error:
```
Structural error at paths./api/contracts.get.parameters.0
  should NOT have additional properties
  additionalProperty: collectionFormat, items
  Jump to line 37`
```

the correct format should be:
```yaml
parameters:
    - name: supply_centers_ids
      in: query
      schema:
        type: array
        items:
          type: integer
```
but if I write my params in the correct form:
```ruby
parameter name: :supply_centers_ids, in: :query, schema: { type: :array, items: { type: :integer } }
```

Rspec goes on error:
```
Failure/Error: return "#{name}=#{value}" unless param[:type].to_sym == :array    
  NoMethodError:
  undefined method `to_sym' for nil:NilClass
```

**Comment by redvex** [comment]
I've dug a bit more on the last error and it seems that the library alters the params after the first example for the query params.

In another spec, I have two different examples of the same request:

```ruby
  path '/api/contracts/{contract_id}/working_days' do
    get 'Retrieves the contract for a specific range' do
      tags WORKING_DAYS_TAG
      produces 'application/json'

      parameter name: :contract_id, in: :path, type: :string
      parameter name: :wc_date, in: :query, type: :string

      response '200', 'elements found' do
        schema '$ref' => '#/components/schemas/working_day_list'
        include_context 'with integration test'
      end

      response '401', 'Unauthorized' do
        schema '$ref' => '#/components/schemas/unautorized_errors'
        let!(:current_user) { build(:user, :teacher)}
        include_context 'with integration test'
      end
    end
  end
```

With `config.swagger_dry_run = false` the second test fails with:
```
Failures:

  1) Api::WorkingDays /api/contracts/{contract_id}/working_days get Unauthorized returns a 401 response
     Got 0 failures and 2 other errors:
     Shared Example Group: "with integration test" called from ./spec/requests/api/working_days_spec.rb:35

     1.1) Failure/Error: return "#{name}=#{value}" unless param[:type].to_sym == :array
          
          NoMethodError:
            undefined method `to_sym' for nil:NilClass
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb:122:in `build_query_string_part'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb:114:in `block (2 levels) in add_path'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb:112:in `each'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb:112:in `each_with_index'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb:112:in `block in add_path'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb:107:in `tap'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb:107:in `add_path'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb:20:in `block in build_request'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb:18:in `tap'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb:18:in `build_request'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/example_helpers.rb:10:in `submit_request'
          # /usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/example_group_helpers.rb:91:in `block in run_test!'
```

I've added a couple of line of debug in `/usr/local/bundle/gems/rswag-specs-2.3.1/lib/rswag/specs/request_factory.rb` and the param hash changes:

1st request: `{:name=>:wc_date, :in=>:query, :type=>:string}`
2nd request: `{:name=>:wc_date, :in=>:query, :schema=>{:type=>:string}}`

if I run the spec with `--dry-run` all works fine, but the examples are not added.

**Comment by diachini** [comment]
@redvex I ran into the same problem you've documented here. If you wanna try out the bandaid I've put up in #319 - give this a shot:

```
  gem 'rswag-specs', git: 'git@github.com:jetpackworkflow/rswag.git', branch: 'allow_oas3_param_schema_array'
```

And see if you have any luck 🤞

**Comment by JorgeGarciaxyz** [comment]
I have the same problem while defining `type` and `default` values
```
page_size: {
  name: :page_size,
  in: :query,
  type: :integer,
  default: 20
}
```

produces
```yaml
  page_size:
    name: page_size
    in: query
    type: integer
    default: 20
```

and should be
```yaml
  page_size:
    name: page_size
    in: query
    schema:
      type: integer
      default: 20
```

**Comment by smeevil** [comment]
Hi @diachini,

I can confirm that your bandaid is working as advertised!
Thank you so much :)

Gerard.

**Comment by seanstory** [comment]
I'm also seeing this issue, and can also validate the fix from #319 . What's the ETA and target version for this fix?

**Comment by seanstory** [comment]
@BookOfGreg is there anything blocking this being merged?

**Comment by jorge0136** [comment]
I too am experiencing this issue, using the `instacart` [fork](https://github.com/instacart/rswag/pull/3) resolves the issue. Is there something we can help with to get this merged @BookOfGreg ?

**Comment by seanstory** [comment]
This seems to be fixed in rswag version 2.4.0
